Tools/Scripts/libraries/reporelaypy/reporelaypy/checkout.py

```diff
@@ -60,19 +61,27 @@ def from_json(cls, data):
         return cls(**data, primary=False)
 
     @staticmethod
-    def clone(url, path, sentinal_file=None):
+    def clone(url, path, remotes, sentinal_file=None):
         run([local.Git.executable(), 'clone', url, path], cwd=os.path.dirname(path))
         run([local.Git.executable(), 'config', 'pull.ff', 'only'], cwd=path)
 
+        Checkout.add_remotes(local.Git(path), remotes)
+
         if sentinal_file:
             with open(sentinal_file, 'w') as cloned:
                 cloned.write('yes\n')
         return 0
 
-    def __init__(self, path, url=None, http_proxy=None, sentinal=True, fallback_url=None, primary=True):
+    @staticmethod
+    def add_remotes(repository, remotes):
+        for name, url in (remotes or {}).items():
+            run([repository.executable(), 'remote', 'add', name, url], cwd=repository.root_path)
+
+    def __init__(self, path, url=None, http_proxy=None, sentinal=True, fallback_url=None, primary=True, remotes=None):
         self.sentinal = sentinal
         self.path = path
         self.url = url
+        self.remotes = dict() or remotes
```

Comment:
Do you mean `self.remotes = remotes or dict()`?

Reply:
Yes, this is a pretty silly bug, will fix.